src/main/webapp/app/course/learning-paths/learning-path-management/learning-path-management.component.ts

```diff
@@ -0,0 +1,207 @@
+            this.courseId = params['courseId'];
+            if (this.courseId) {
+                this.loadData();
+            }
+        });
+    }
+
+    private loadData() {
+        this.isLoading = true;
+
+        this.learningPathService
+            .getHealthStatusForCourse(this.courseId)
+            .pipe(
+                finalize(() => {
+                    this.isLoading = false;
+                }),
+            )
+            .subscribe({
+                next: (res) => {
+                    this.health = res.body!;
+                    if (this.health.status !== HealthStatus.DISABLED) {
+                        this.performSearch(this.sort, 0);
+                        this.performSearch(this.search, 300);
+                    }
+                },
+                error: (res: HttpErrorResponse) => onError(this.alertService, res),
+            });
+    }
+
+    enableLearningPaths() {
+        this.isLoading = true;
+        this.learningPathService.enableLearningPaths(this.courseId).subscribe({
+            next: () => {
+                this.loadData();
+            },
+            error: (res: HttpErrorResponse) => onError(this.alertService, res),
+        });
+    }
+
+    generateMissing() {
+        this.isLoading = true;
+        this.learningPathService.generateMissingLearningPathsForCourse(this.courseId).subscribe({
+            next: () => {
+                this.loadData();
+            },
+            error: (res: HttpErrorResponse) => onError(this.alertService, res),
+        });
+    }
+
+    /**
+     * Method to perform the search based on a search subject
+     *
+     * @param searchSubject The search subject which we use to search.
+     * @param debounce The delay we apply to delay the feedback / wait for input
+     */
+    performSearch(searchSubject: Subject<void>, debounce: number): void {
+        searchSubject
+            .pipe(
+                debounceTime(debounce),
+                tap(() => (this.searchLoading = true)),
+                switchMap(() => this.pagingService.searchForLearningPaths(this.state, this.courseId)),
+            )
+            .subscribe((resp) => {
+                this.content = resp;
+                this.searchLoading = false;
+                this.total = resp.numberOfPages * this.state.pageSize;
+            });
+    }
+
+    sortRows() {
+        this.sortService.sortByProperty(this.content.resultsOnPage, this.sortedColumn, this.listSorting);
+    }
+
+    private setSearchParam(patch: Partial<PageableSearch>): void {
+        Object.assign(this.state, patch);
+        this.sort.next();
+    }
+
+    /**
+     * Callback function when the user navigates through the page results
+     *
+     * @param pageNumber The current page number
+     */
+    onPageChange(pageNumber: number) {
+        if (pageNumber) {
+            this.page = pageNumber;
+        }
+    }
+    viewLearningPath(learningPath: LearningPathPageableSearchDTO) {
```

Comment:
client-style warning.

Reply:
This method is a placeholder. It is not an issue in #7027 that will be merged immediately after this PR. If you prefer I can remove the argument here and add it again in the next PR. Just let me know.

Reply:
That's fine for me.